api/envoy/service/ratelimit/v3/rls.proto

```diff
@@ -118,6 +119,17 @@ message RateLimitResponse {
     google.protobuf.Duration duration_until_reset = 4;
   }
 
+  // Cacheable quota for responses, see documentation for the :ref:`quota
+  // <envoy_v3_api_field_service.ratelimit.v3.RateLimitResponse.quota>` field.
+  // [#not-implemented-hide:]
+  message Quota {
+    // Number of matching requests granted in quota.
+    uint32 requests = 1;
+
+    // Point in time at which the quota expires.
+    google.protobuf.Timestamp valid_until = 2;
```

Comment:
Perhaps throw this in a oneof? I could see someone wanting to do a duration upon receipt to avoid having to rely on clock sync?